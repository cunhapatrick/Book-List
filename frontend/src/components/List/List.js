import React from 'react';
import PropTypes from 'prop-types';

// CSS
import {
  Container,
} from './List.style';

const List = ({ books, handleDetalhes }) => {
  return (
  <Container>
    <table>
      <thead>
        <tr>
          <th>Livro</th>
          <th>Autor</th>
          <th>Editora</th>
          <th>Ano</th>
          <th>Ações</th>
        </tr>
      </thead>
      <tbody>
        {books.map(book => (
          <tr key={book._id}>
            <td>{book.title}</td>
            <td>{book.author}</td>
            <td>{book.publisher}</td>
            <td>{book.year}</td>
            <td>
              <button onClick={() => handleDetalhes(book)}>Detalhes</button>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  </Container>
);
}
List.propTypes = {
  books: PropTypes.array.isRequired,
  handleDetalhes: PropTypes.func.isRequired,
}

export default List;