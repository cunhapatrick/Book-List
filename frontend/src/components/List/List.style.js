import styled from 'styled-components';

export const Container = styled.div`
  margin: 30px;
  table {
    border: 1px solid black;
    border-collapse: collapse;
    width: 100%;
    
    td, th {
      border-left: 1px solid black;
    }

    td {
      padding: 5px;
    }

    th {
      height: 50px;
      text-align: left;
      background-color: #4CAF50;
      color: white;
      text-align: center;
    }

    tr {
      &:nth-child(even) {background-color: #f2f2f2;}
      &:hover {
        background-color: #f5f5f5;
      }
      > td {
        text-align: center;
      }
    }

    button {
      border: none;
      background: inherit;
      text-decoration: underline;
      color: #5F00FF
    }

  }
`;