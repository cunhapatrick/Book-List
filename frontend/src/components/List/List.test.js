import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('Testing List Component', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <List
        books={[{
          _id: 1,
          title: 'teste',
          author: 'teste',
        }]}
        handleDetalhes={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});