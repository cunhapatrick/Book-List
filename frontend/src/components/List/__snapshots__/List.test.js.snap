// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing List Component should render properly 1`] = `ShallowWrapper {}`;
